From 55e02c15b03009287be5ed3dc3f259e4d3635c0a Mon Sep 17 00:00:00 2001
From: Fabrice Goucem <fabrice.goucem@nxp.com>
Date: Thu, 10 Mar 2022 10:55:49 +0100
Subject: [PATCH 11/28] configs: arm64: imx8m{m,n,p}: Add all SAI instances to
 harpoon audio cell config

[HRPN-293] imx8m{m,n,p}-freertos-audio: fix permissions for SAI and I2C

All SAI and I2C3 should not be accessed by Linux.

Signed-off-by: Fabrice Goucem <fabrice.goucem@nxp.com>
---
 configs/arm64/imx8mm-harpoon-freertos-audio.c | 17 ++++++++++++-----
 configs/arm64/imx8mm-harpoon-rtos.h           |  3 +++
 configs/arm64/imx8mn-harpoon-freertos-audio.c | 14 ++++++++++----
 configs/arm64/imx8mn-harpoon-rtos.h           |  2 ++
 configs/arm64/imx8mp-harpoon-freertos-audio.c | 17 ++++++++++++-----
 configs/arm64/imx8mp-harpoon-rtos.h           |  3 +++
 6 files changed, 42 insertions(+), 14 deletions(-)

diff --git a/configs/arm64/imx8mm-harpoon-freertos-audio.c b/configs/arm64/imx8mm-harpoon-freertos-audio.c
index dff87033..29bc5a13 100644
--- a/configs/arm64/imx8mm-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mm-harpoon-freertos-audio.c
@@ -28,8 +28,12 @@
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
 	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(64)),   /* ANA_PLL */\
 	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
-	MMIO_REGION_RWS(0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */   \
-	MMIO_REGION_RWS(0x30050000, 0x30050000, KB(64))    /* SAI5 */
+	MMIO_REGION_RW( 0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */   \
+	MMIO_REGION_RW( 0x30010000, 0x30010000, KB(64)),   /* SAI1 */   \
+	MMIO_REGION_RW( 0x30020000, 0x30020000, KB(64)),   /* SAI2 */   \
+	MMIO_REGION_RW( 0x30030000, 0x30030000, KB(64)),   /* SAI3 */   \
+	MMIO_REGION_RW( 0x30050000, 0x30050000, KB(64)),   /* SAI5 */   \
+	MMIO_REGION_RW( 0x30060000, 0x30060000, KB(64))    /* SAI6 */   \
 
 /* GIC */
 #define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
@@ -39,12 +43,15 @@
 		RTOS_COMMON_IRQCHIP0_BITMAP1,              \
 		/* interrupts 64..95 */                    \
 		RTOS_COMMON_IRQCHIPS_BITMAP2 |             \
-		IRQnMAP(I2C3_IRQn),  /* I2C3 */            \
+		IRQnMAP(I2C3_IRQn) | /* I2C3 */            \
+		IRQnMAP(I2S3_IRQn),  /* SAI3 */            \
 		/* interrupts 96..127 */                   \
 		RTOS_COMMON_IRQCHIPS_BITMAP3 |             \
-		IRQnMAP(I2S56_IRQn), /* SAI5 */            \
+		IRQnMAP(I2S56_IRQn) | /* SAI5-6 */         \
+		IRQnMAP(I2S1_IRQn),   /* SAI1 */           \
 		/* interrupts 128..159 */                  \
-		RTOS_COMMON_IRQCHIPS_BITMAP4)
+		RTOS_COMMON_IRQCHIPS_BITMAP4 |             \
+		IRQnMAP(I2S2_IRQn))   /* SAI2 */
 
 /* IVSHMEM PCI */
 #define CONFIG_INMATE_PCI_DEVICES \
diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index 4acd1322..ccc8e80e 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -17,9 +17,12 @@ typedef enum IRQn {
   /* Device specific interrupts */
   UART4_IRQn                   = 61,               /**< UART-4 ORed interrupt */
   I2C3_IRQn                    = 69,               /**< I2C-3 Interrupt */
+  I2S3_IRQn                    = 82,               /**< SAI3 Receive / Transmit Interrupt */
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  I2S1_IRQn                    = 127,              /**< SAI1 Receive / Transmit Interrupt */
+  I2S2_IRQn                    = 128,              /**< SAI2 Receive / Transmit Interrupt */
   ENET_MAC0_Rx_Tx_Done1_IRQn   = 150,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
   ENET_MAC0_Rx_Tx_Done2_IRQn   = 151,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
   ENET_IRQn                    = 152,              /**< MAC 0 IRQ */
diff --git a/configs/arm64/imx8mn-harpoon-freertos-audio.c b/configs/arm64/imx8mn-harpoon-freertos-audio.c
index d553d201..9d9898aa 100644
--- a/configs/arm64/imx8mn-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mn-harpoon-freertos-audio.c
@@ -28,8 +28,12 @@
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
 	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(64)),   /* ANA_PLL */\
 	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
-	MMIO_REGION_RWS(0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */   \
-	MMIO_REGION_RWS(0x30050000, 0x30050000, KB(64))    /* SAI5 */
+	MMIO_REGION_RW( 0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */   \
+	MMIO_REGION_RW( 0x30020000, 0x30020000, KB(64)),   /* SAI2 */   \
+	MMIO_REGION_RW( 0x30030000, 0x30030000, KB(64)),   /* SAI3 */   \
+	MMIO_REGION_RW( 0x30050000, 0x30050000, KB(64)),   /* SAI5 */   \
+	MMIO_REGION_RW( 0x30060000, 0x30060000, KB(64)),   /* SAI6 */   \
+	MMIO_REGION_RW( 0x300b0000, 0x300b0000, KB(64))    /* SAI7 */
 
 /* GIC */
 #define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
@@ -43,9 +47,11 @@
 		IRQnMAP(I2S3_IRQn),  /* SAI3 */            \
 		/* interrupts 96..127 */                   \
 		RTOS_COMMON_IRQCHIPS_BITMAP3 |             \
-		IRQnMAP(I2S56_IRQn),  /* SAI5 */           \
+		IRQnMAP(I2S56_IRQn),  /* SAI5-6 */         \
 		/* interrupts 128..159 */                  \
-		RTOS_COMMON_IRQCHIPS_BITMAP4)
+		RTOS_COMMON_IRQCHIPS_BITMAP4 |             \
+		IRQnMAP(I2S2_IRQn) |  /* SAI2 */           \
+		IRQnMAP(I2S7_IRQn))   /* SAI7 */
 
 /* IVSHMEM PCI */
 #define CONFIG_INMATE_PCI_DEVICES \
diff --git a/configs/arm64/imx8mn-harpoon-rtos.h b/configs/arm64/imx8mn-harpoon-rtos.h
index dca81931..47aeab01 100644
--- a/configs/arm64/imx8mn-harpoon-rtos.h
+++ b/configs/arm64/imx8mn-harpoon-rtos.h
@@ -21,6 +21,8 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  I2S2_IRQn                    = 128,              /**< SAI2 Receive / Transmit Interrupt */
+  I2S7_IRQn                    = 143,              /**< SAI7 Receive / Transmit Interrupt */
   ENET_MAC0_Rx_Tx_Done1_IRQn   = 150,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
   ENET_MAC0_Rx_Tx_Done2_IRQn   = 151,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
   ENET_IRQn                    = 152,              /**< MAC 0 IRQ */
diff --git a/configs/arm64/imx8mp-harpoon-freertos-audio.c b/configs/arm64/imx8mp-harpoon-freertos-audio.c
index 9d060d45..89d06d3b 100644
--- a/configs/arm64/imx8mp-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mp-harpoon-freertos-audio.c
@@ -32,9 +32,13 @@
 	MMIO_REGION_RWS(0x30384000, 0x30384000, KB(4)),    /* CCM */    \
 	MMIO_REGION_RWS(0x30388000, 0x30388000, KB(12)),   /* CCM */    \
 	MMIO_REGION_RW( 0x303a0000, 0x303a0000, KB(64)),   /* GPC */	\
-	MMIO_REGION_RWS(0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */	\
-	MMIO_REGION_RWS(0x30c30000, 0x30c30000, KB(64)),   /* SAI3 */	\
-	MMIO_REGION_RWS(0x30c50000, 0x30c50000, KB(64)),   /* SAI5 */	\
+	MMIO_REGION_RW( 0x30a40000, 0x30a40000, KB(64)),   /* I2C3 */	\
+	MMIO_REGION_RW( 0x30c10000, 0x30c10000, KB(64)),   /* SAI1 */	\
+	MMIO_REGION_RW( 0x30c20000, 0x30c20000, KB(64)),   /* SAI2 */	\
+	MMIO_REGION_RW( 0x30c30000, 0x30c30000, KB(64)),   /* SAI3 */	\
+	MMIO_REGION_RW( 0x30c50000, 0x30c50000, KB(64)),   /* SAI5 */	\
+	MMIO_REGION_RW( 0x30c60000, 0x30c60000, KB(64)),   /* SAI6 */	\
+	MMIO_REGION_RW( 0x30c80000, 0x30c80000, KB(64)),   /* SAI7 */	\
 	MMIO_REGION_RWS(0x30e20000, 0x30e20000, KB(64))    /* AUDIOMIX */
 
 /* GIC */
@@ -48,9 +52,12 @@
 		IRQnMAP(I2S3_IRQn),  /* SAI3 */            \
 		/* interrupts 96..127 */                   \
 		RTOS_COMMON_IRQCHIPS_BITMAP3 |             \
-		IRQnMAP(I2S56_IRQn), /* SAI5 */            \
+		IRQnMAP(I2S56_IRQn) | /* SAI5-6 */         \
+		IRQnMAP(I2S1_IRQn),   /* SAI1 */           \
 		/* interrupts 128..159 */                  \
-		RTOS_COMMON_IRQCHIPS_BITMAP4)
+		RTOS_COMMON_IRQCHIPS_BITMAP4 |             \
+		IRQnMAP(I2S2_IRQn) |  /* SAI2 */           \
+		IRQnMAP(I2S7_IRQn))   /* SAI7 */
 
 /* IVSHMEM PCI */
 #define CONFIG_INMATE_PCI_DEVICES \
diff --git a/configs/arm64/imx8mp-harpoon-rtos.h b/configs/arm64/imx8mp-harpoon-rtos.h
index 47cb99f1..f82a811a 100644
--- a/configs/arm64/imx8mp-harpoon-rtos.h
+++ b/configs/arm64/imx8mp-harpoon-rtos.h
@@ -20,6 +20,9 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  I2S1_IRQn                    = 127,              /**< SAI1 Receive / Transmit Interrupt */
+  I2S2_IRQn                    = 128,              /**< SAI2 Receive / Transmit Interrupt */
+  I2S7_IRQn                    = 143,              /**< SAI7 Receive / Transmit Interrupt */
   ENET_QOS_PMT_IRQn            = 166,              /**< ENET QOS TSN Interrupt from PMT */
   ENET_QOS_IRQn                = 167,              /**< ENET QOS TSN LPI RX exit/Host System/RX/TX Channels[4:0] Interrupt */
   CAN_FD1_IRQn                 = 174,              /**< CAN-FD1 Interrupt from bus off/line error/RX warning/TX warning/wakeup/match in PN/timeout in PN/busoff done/FD error */
-- 
2.34.1

